nautobot/extras/navigation.py

```diff
@@ -391,8 +391,10 @@
                         weight=100,
                         permissions=[],
                         buttons=(),
+                        admin_only=True,
                     ),
                 ),
+                admin_only=True,
```

Comment:
For readability, would you mind moving this to before the `items` parameter?

Actually, does the NavMenuGroup *need* to be `admin_only=True`? What if a plugin wants to add a non-admin item to this group? I'm guessing this was added because otherwise we get an empty `General` group appearing in the menu for non-admin users - maybe that's a bug in the template rendering logic that we should fix instead?